cmd/snap-confine/mount-support.c

```diff
@@ -603,11 +608,11 @@ void sc_populate_mount_ns(struct sc_apparmor *apparmor, int snap_update_ns_fd,
 	if (vanilla_cwd == NULL) {
 		die("cannot get the current working directory");
 	}
-
-	bool on_classic_distro = is_running_on_classic_distribution();
-	// on classic or with alternative base snaps we need to setup
-	// a different confinement
-	if (on_classic_distro || !sc_streq(base_snap_name, "core")) {
+	// Classify the current distribution, as claimed by /etc/os-release.
+	sc_distro distro = sc_classify_distro();
+	// Check which mode we should run in, normal or legacy.
+	if (sc_should_use_normal_mode(distro, base_snap_name)) {
```

Comment:
What if we reversed the logic, and had `sc_should_use_legacy_mode(distro, base_snap_name)` thus avoiding 'normal mode' and the confusion?